src/autodiff_model.jl

```diff
@@ -242,6 +242,32 @@ function hprod!(nlp :: ADNLPModel, x :: AbstractVector, y :: AbstractVector, v :
   return Hv
 end
 
+function jth_hess_coord!(nlp :: ADNLPModel, x :: AbstractVector, j :: Integer, vals :: AbstractVector)
+  @lencheck nlp.meta.nnzh vals
+  @lencheck nlp.meta.nvar x
+  @assert 1 ≤ j ≤ nlp.meta.ncon
+  increment!(nlp, :neval_jhess)
+  ℓ(x) = nlp.c(x)[j]
+  Hx = ForwardDiff.hessian(ℓ, x)
+  k = 1
+  for j = 1 : nlp.meta.nvar
+    for i = j : nlp.meta.nvar
+      vals[k] = Hx[i, j]
+      k += 1
+    end
+  end
+  return vals
+end
+
+function jth_hprod!(nlp :: ADNLPModel, x :: AbstractVector, v :: AbstractVector, j :: Integer, Hv :: AbstractVector)
+  @lencheck nlp.meta.nvar x v Hv
+  @assert 1 ≤ j ≤ nlp.meta.ncon
```

Comment:
```suggestion
  @rangecheck 1 nlp.meta.ncon j
```
I'm not sure of the syntax, but we have `@rangecheck` in the same file as `@lencheck`.